src/components/AnimatePresence.tsx

```diff
@@ -79,8 +79,20 @@ function updateChildLookup(
     children: ReactElement<any>[],
     allChildren: Map<ComponentKey, ReactElement<any>>
 ) {
+    const seenChildren =
+        process.env.NODE_ENV !== "production" ? new Set<ComponentKey>() : null
```

Comment:
those env checks look slightly weird but they provide better DCE in production builds